Riot/Managers/KeyValueStorage/KeychainStore.swift

```diff
@@ -0,0 +1,47 @@
+// 
+// Copyright 2020 Vector Creations Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+import KeychainAccess
+
+@objcMembers
+/// Only supports `String` and `Data` values for now.
+class KeychainStore: KeyValueStore {
```

Comment:
Regarding the KeychainStore:

- The keychain can be accessible from share extension. We should define the behavior in this case ask to go in the app or present pin code from share extension. So we will probably instantiate like this:
`let keychain = Keychain(service: "com.example.github-token", accessGroup: "12ABCD3E4F.shared")`
- We should handle errors to distinct the case there is no value and the case there is an error while accessing value espcecially to be sure the data is stored after setting the pin code for example. The Keychain data manipulation can fail, like seen in `KeychainAccess` interface:
`public func set(_ value: String, key: String) throws`
- To control keys and avoid mistakes `KeyValueStoreKey` should probably be an enum with String representation:
```
enum KeyValueStoreKey: String {
	case pinCode = "pincode"
}
```
- Instead of `func setObject(forKey key: KeyValueStoreKey, value: Any?)` we can use the same as `KeychainAccess`: `func set(_ value: Any?, key: KeyValueStoreKey) throws`
- We should handle user disconnection and provide a `removeAll()` method
- If we want to constraint an item with Face ID we will to handle it in asynchronous way: https://github.com/kishikawakatsumi/KeychainAccess#-cyclone-touch-id-face-id-integration. It just to say even if KeyValueStore is a great idea it will probably don't fill every needs :/